include/swift/ABI/Metadata.h

```diff
@@ -1495,7 +1495,7 @@ struct TargetTupleTypeMetadata : public TargetMetadata<Runtime> {
     ConstTargetMetadataPointer<Runtime, swift::TargetMetadata> Type;
 
     /// The offset of the tuple element within the tuple.
-    StoredSize Offset;
+    uint32_t Offset;
```

Comment:
Looks good! For ABI compatibility on Apple platforms, we'd want to conditionalize this, because existing binaries will try to load the full 64 bits, and we want to keep those zeroed:

```
#if __APPLE__
  StoredSize Offset;
#else
  uint32_t Offset;
#endif
```

I think the IRGen side is fine, though, because in practice nobody is going to have a 4GB tuple (they'll run into other practical limits first), and new binaries doing a 32-bit little-endian load of the 64-bit field should be safe, and would open the door to us possibly being able to drop this compatibility hack in time.

Reply:
@jckarter Change made. Thanks.